### PR TITLE
Support / LIVE-7191: bump react-native-reanimated to v3

### PR DESCRIPTION
**Pull request by alexandremgo**
### 📝 Description

Our usage of [react-native-reanimated](https://github.com/software-mansion/react-native-reanimated) in v2.15 was creating freezing issues on iOS (see ticket).

### LLM

- Bumped [react-native-reanimated](https://github.com/software-mansion/react-native-reanimated) to v3.1.0.
- Added necessary babel plugins to `devDependencies` that are needed for `react-native-reanimated`.
- Also updated `metro.config.js` to stop forcing resolving react-native-reanimated from the LLM folder as it was creating import issues.
- Bumped [lottie-react-native](https://github.com/lottie-react-native/lottie-react-native) to `6.0.0-rc.6`
  - Needed because with Reanimated 3, `Lottie` from `lottie-react-native v5.1.5` were not working correctly anymore

👷 Also adapted a bunch of components that were still using the v1 api of reanimated -> I wrote a plan for QA to double check them

🧹🧼🧽 After adapting those components, removed the following unused components/libs:
- `react-native-redash` dependency : unused
- `lottie-ios` dependency: unneeded since `lottie-react-native` > 5.1.4  
- some components that were used by an old onboarding that cannot be reach anymore:
  - `apps/ledger-live-mobile/src/components/AnimatedSvgBackground.tsx`
  - `apps/ledger-live-mobile/src/components/OnboardingStepperView/OnboardingStepView.tsx`
  - `apps/ledger-live-mobile/src/components/OnboardingStepperView/index.tsx`

### native-ui

- Bumped [react-native-reanimated](https://github.com/software-mansion/react-native-reanimated) to v3.1.0.

It looks like all the components from the native-ui lib that are using `react-native-reanimated` were using the v2 apis of the libs already (not the v1 api like some components in LLM) and there is no code/api errors that i see

But we should double check the storybooks
<details>
<summary>The native-ui components using react-native-reanimated</summary>

To check on the storybook -> check them once you see them

- [x] : `libs/ui/packages/native/src/components/Layout/Collapse/Accordion/index.tsx`
- [x] : `libs/ui/packages/native/src/components/Layout/List/VerticalStepper/StepperItem.tsx`
- [x] : `libs/ui/packages/native/src/components/Layout/List/VerticalTimeline/TimelineItem.tsx`
- [ ] : `libs/ui/packages/native/src/components/Layout/ScrollContainer/index.tsx`
- [ ] : `libs/ui/packages/native/src/components/Layout/ScrollContainerHeader/Header.tsx`
- [ ] : `libs/ui/packages/native/src/components/Layout/ScrollContainerHeader/index.tsx`
- [ ] : `libs/ui/packages/native/src/components/Layout/ScrollListContainer/index.tsx`
- [x] : `libs/ui/packages/native/src/components/Loader/InfiniteLoader/index.tsx`
- [ ] : `libs/ui/packages/native/src/components/Loader/ProgressLoader/index.tsx`
- [ ] : `libs/ui/packages/native/src/components/Navigation/SlideIndicator/index.tsx`
- [ ] : `libs/ui/packages/native/src/components/Navigation/Stepper/index.tsx`
- [ ] : `libs/ui/packages/native/src/components/Navigation/StoriesIndicator/index.tsx`
- [ ] : `libs/ui/packages/native/src/components/ProgressBar/index.tsx`

</details>
### ❓ Context

- **Impacted projects**: `LLM`
- **Linked resource(s)**: [LIVE-7191](https://ledgerhq.atlassian.net/browse/LIVE-7191)
### ✅ Checklist

- [ ] **Test coverage** 
- [x] **Atomic delivery** 
- [ ] **No breaking changes** 

### 📸 Demo

<details>
  <summary>Demo on iOS: full</summary>

https://user-images.githubusercontent.com/15096913/236474287-88f9d28a-b5c1-425e-9833-7fe79e6d8723.mp4

</details>

<details>
  <summary>Demo on Android: full</summary>

https://user-images.githubusercontent.com/15096913/236474385-f9bf04d3-cd74-4937-808f-415bcd319259.mp4

</details>

<details>
  <summary>Demo on Android: toggle button in Cosmos flow</summary>

https://github.com/LedgerHQ/ledger-live/assets/15096913/f6706e5d-11c5-4019-a25a-012c4d8fdd5f

</details>

</details>

<details>
  <summary>Demo on Android: Snackbar/toasts </summary>

https://github.com/LedgerHQ/ledger-live/assets/15096913/47c8f201-078d-4461-90de-fe5260f37ccf

</details>

<details>
  <summary>Demo on Android: Discover v2 still working</summary>

https://github.com/LedgerHQ/ledger-live/assets/15096913/ce2d9678-2517-4c29-af4e-568aca5da7d4

</details>

<details>
  <summary>Demo on Android: progress bar</summary>

https://github.com/LedgerHQ/ledger-live/assets/15096913/358313cd-3ad4-4074-869b-954a8a6301ae

</details>

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-7191]: https://ledgerhq.atlassian.net/browse/LIVE-7191?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by elbywan — APPROVED** [review]

**Review by cgrellard-ledger — APPROVED** [review]
LGTM !

File: apps/ledger-live-mobile/src/components/TabBar/Transfer.tsx [hunk 1]
```diff
@@ -120,6 +122,15 @@ export function TransferTabIcon() {
...
 
+  /**
+   * openAnimValue.value:
+   *            0             ->       1           ->             2
+   * transfer arrow icon (0)     close icon (0.5)        transfer arrow icon (1)
+   * with intermediate animation/steps in decimal (test 0.2 for ex)
+   *
+   * progress: from lottie-react-native. Represents the normalized progress of the animation.
+   * If this prop is updated, the animation will correspondingly update to the frame at that progress value.
+   */
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/components/TabBar/Transfer.tsx:133]: great, thanks for documenting it!

File: apps/ledger-live-mobile/src/screens/Onboarding/shared/infoPagesData.tsx [hunk 2]
```diff
@@ -469,552 +461,3 @@ export const infoModalScenes = {
   hideRecoveryPhraseInfoModalProps,
   pairNewErrorInfoModalProps,
 };
-
-const pinCodeScenes = (
```
Comment (alexandremgo) [inline apps/ledger-live-mobile/src/screens/Onboarding/shared/infoPagesData.tsx:473]: Great 🧨

Comment (ofreyssinet-ledger) [reply apps/ledger-live-mobile/src/screens/Onboarding/shared/infoPagesData.tsx:473]: kill it with fire
also this lottieAnimations at the top is only used in a debug screen so I think there's something weird there

**Review by ofreyssinet-ledger — APPROVED** [review]
lgtm 👍  great great job, thanks for taking care of that !
I added some cleanup (unused code & files), slightly refactored the Lottie debug screen and also refactored the SnackBar solution with a nonce (the nonce logic is now implemented in the ToastProvider rather than in a useEffect with setState which caused an extra unnecessary re-render)

File: libs/ledger-live-common/src/notifications/ToastProvider/index.tsx [hunk 3]
```diff
@@ -23,21 +25,25 @@ export function useToasts(): ToastContextType {
 
 export function ToastProvider({ children }: Props): ReactElement {
   const [toasts, setToasts] = useState<ToastData[]>([]);
+  const [nonce, setNonce] = useState(0);
```
Comment (alexandremgo) [inline libs/ledger-live-common/src/notifications/ToastProvider/index.tsx:28]: Nice, thx !

**Pull request merged**